### PR TITLE
Update kubeadm v1alpha3 example configuration

### LINKED ISSUE #1132: Restructure v1alpha3 godoc example
Updating the example is all that is required here

### PR DESCRIPTION
**Pull request by chuckha**
closes kubernetes/kubeadm#1132

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

**What this PR does / why we need it**:
This updates the godoc example.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#1132

**Special notes for your reviewer**:
This is easy to typo and has no test/validation as it's a comment. Please double check typos.

```release-note
NONE
```

/assign @timothysc 
/cc @liztio @rosti @neolit123

### CODE REVIEW

**Review by timothysc — APPROVED** [review]
/approve

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 1]
```diff
@@ -52,41 +76,38 @@ limitations under the License.
...
+//  - name: "some-volume"
+//    hostPath: "/etc/some-path"
+//    mountPath: "/etc/some-pod-path"
+//    writable: true
+//    pathType: File
 //	apiServerCertSANs:
 //	- "10.100.1.1"
 //	- "ec2-10-100-0-1.compute-1.amazonaws.com"
-//	certificateDirectory: "/etc/kubernetes/pki"
+//	certificatesDir: "/etc/kubernetes/pki"
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: [1]

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 2]
```diff
@@ -52,41 +76,38 @@ limitations under the License.
...
 //	imageRepository: "k8s.gcr.io"
 //	unifiedControlPlaneImage: "k8s.gcr.io/controlplane:v1.12.0"
-//	auditPolicyConfiguration:
+//	auditPolicy:
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: [1]
yep, these names do not match the struct types and are misleading. :\
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1alpha3/types.go#L116

my vote would be to make them match.

**Review by neolit123 — COMMENTED** [review]
LGTM, thanks.

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 3]
```diff
@@ -52,41 +76,38 @@ limitations under the License.
...
+//    mountPath: "/etc/some-pod-path"
+//    writable: true
+//    pathType: File
+//  schedulerExtraVolumes:
+//  - name: "some-volume"
+//    hostPath: "/etc/some-path"
+//    mountPath: "/etc/some-pod-path"
+//    writable: true
+//    pathType: File
 //	apiServerCertSANs:
```
Comment (liztio) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: this sort of looks like it should be indented one level less?

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 4]
```diff
@@ -16,12 +16,36 @@ limitations under the License.
 
 // Package v1alpha3 is the API (config file) for driving the kubeadm binary.
 // Some of these options are also available as command line flags, but
-// the preferred way to configure kubeadm is to pass a YAML file in with the
-// --config option.
+// the preferred way to configure kubeadm is to pass a single YAML file with
+// multiple configuration types in with the --config option.
```
Comment (rosti) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: Although in the future the users may have the choice to store different `kind`s in different YAML files, currently we don't. So this must be single YAML file and we should emphasize on that here.
Also, InitConfiguration and JoinConfiguration cannot share a single YAML file at the moment (kubeadm bails out). So we should probably mention that too. Not sure for ClusterConfiguration and JoinConfiguration though.

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go (hunk 3 above)
Comment (liztio) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: Actually this whole section's indentation is off

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 5]
```diff
@@ -16,12 +16,36 @@ limitations under the License.
...
 //
-// A fully populated example of the schema:
+// A fully populated example of a configuration file:
```
Comment (rosti) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: Perhaps `s/a configuration/an init configuration` ?

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go (hunk 2 above)
Comment (liztio) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: good catch on these two

**Review by liztio — CHANGES_REQUESTED** [review]

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go (hunk 4 above)
Comment (neolit123) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: > Not sure for ClusterConfiguration and JoinConfiguration though.

yeah, we probably want to handle all scenarios.
to exclude the possibility of related `k/kubeadm` issue reports, that is.

Comment (chuckha) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: i'll add some additional text to catch edge cases.

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go (hunk 5 above)
Comment (chuckha) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: init configuration is a little misleading as we have an object called "InitConfiguration" and "ClusterConfiguration" when a cluster configuration is actually the useful thing to use during `kubeadm init`, but we could use both.

I think configuration file gets the meaning across while avoiding this confusion, but if you feel strongly I can change it.

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go (hunk 3 above)
Comment (chuckha) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: yes, it totally is, good 👀

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go (hunk 5 above)
Comment (rosti) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: My idea here is not to confuse a user, that JoinConfiguration can be appended to this file. So if we manage to cover that case with some kind of word shuffling in either this or the previous paragraph, I am totally fine with it.

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 6]
```diff
@@ -16,12 +16,36 @@ limitations under the License.
...
 //	apiVersion: kubeadm.k8s.io/v1alpha3
 //	kind: InitConfiguration
+//	bootstrapTokens:
+//	- token: "9a08jv.c0izixklcxtmnze7"
+//	  description: "kubeadm bootstrap token"
+//	  ttl: "24h"
+//	  usages:
```
[the remark this answers is not in the document]
Comment (chuckha) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: I'm with you, +1 for sure

File: cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go [hunk 7]
```diff
@@ -16,12 +16,47 @@ limitations under the License.
...
+// multiple configuration types in with the --config option.
 //
-// A fully populated example of the schema:
+// kubeadm defines several configuration types:
+// * InitConfiguration
+// * JoinConfiguration
+// * ClusterConfiguration
+//
+// InitConfiguration and JoinConfiguration cannot share a single YAML file,
+// however it is expected thatInitConfiguration and ClusterConfiguration will
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: space after `that`, before `Init`...

Comment (chuckha) [reply cmd/kubeadm/app/apis/kubeadm/v1alpha3/doc.go]: such good eyes 🦅 👀

**Review by timothysc — APPROVED** [review]
LGMT @neolit123 for final apply

**Comment by neolit123** [comment]
we are probably going to get :question: from users regardless, but this is the right direction.
/lgtm

**Pull request merged**